storage/storage.go

```diff
@@ -308,7 +308,11 @@ type bucketBoundHostname struct {
 }
 
 func (s pathStyle) host(bucket string) string {
-	return "storage.googleapis.com"
+	if os.Getenv("STORAGE_EMULATOR_HOST") {
+		return os.Getenv("STORAGE_EMULATOR_HOST")
```

Comment:
We accept STORAGE_EMULATOR_HOST both with and without a scheme, so we should strip the scheme if it is present. See https://github.com/googleapis/google-cloud-go/blob/main/storage/storage.go#L154